modules/dbparser/tests/test_patterndb.h

```diff
@@ -0,0 +1,538 @@
+       </action>\
+     </actions>\
+    </rule>\
+  </rules>\
+</ruleset>\
+</patterndb>"
+
+#define pdb_inheritance_enabled_skeleton "<patterndb version='4' pub_date='2010-02-22'>\
+  <ruleset name='testset' id='1'>\
+    <patterns>\
+      <pattern>prog2</pattern>\
+    </patterns>\
+    <rules>\
+      <rule provider='test' id='11' class='system'>\
+        <patterns>\
+          <pattern>pattern-with-inheritance-enabled</pattern>\
+        </patterns>\
+        <tags>\
+          <tag>basetag1</tag>\
+          <tag>basetag2</tag>\
+        </tags>\
+        <actions>\
+          <action trigger='match'>\
+            <message inherit-properties='TRUE'>\
+              <values>\
+                <value name='actionkey'>actionvalue</value>\
+              </values>\
+              <tags>\
+                <tag>actiontag</tag>\
+              </tags>\
+            </message>\
+          </action>\
+        </actions>\
+      </rule>\
+    </rules>\
+  </ruleset>\
+</patterndb>"
+
+#define pdb_inheritance_disabled_skeleton "<patterndb version='4' pub_date='2010-02-22'>\
+  <ruleset name='testset' id='1'>\
+    <patterns>\
+      <pattern>prog2</pattern>\
+    </patterns>\
+    <rules>\
+      <rule provider='test' id='12' class='system'>\
+        <patterns>\
+          <pattern>pattern-with-inheritance-disabled</pattern>\
+        </patterns>\
+        <tags>\
+          <tag>basetag1</tag>\
+          <tag>basetag2</tag>\
+        </tags>\
+        <actions>\
+          <action trigger='match'>\
+            <message inherit-properties='FALSE'>\
+              <values>\
+                <value name='actionkey'>actionvalue</value>\
+              </values>\
+              <tags>\
+                <tag>actiontag</tag>\
+              </tags>\
+            </message>\
+          </action>\
+        </actions>\
+      </rule>\
+    </rules>\
+ </ruleset>\
+</patterndb>"
+
+#define pdb_inheritance_context_skeleton "\
+<patterndb version='4' pub_date='2010-02-22'>\
+  <ruleset name='testset' id='1'>\
+    <patterns>\
+      <pattern>prog2</pattern>\
+    </patterns>\
+    <rules>\
+      <rule provider='test' id='11' class='system' context-scope='program'\
+           context-id='$PID' context-timeout='60'>\
+        <patterns>\
+          <pattern>pattern-with-inheritance-context</pattern>\
+        </patterns>\
+        <tags>\
+          <tag>basetag1</tag>\
+          <tag>basetag2</tag>\
+        </tags>\
+        <actions>\
+          <action trigger='timeout'>\
+            <message inherit-properties='context'>\
+              <values>\
+                <value name='MESSAGE'>action message</value>\
+              </values>\
+              <tags>\
+                <tag>actiontag</tag>\
+              </tags>\
+            </message>\
+          </action>\
+        </actions>\
+     </rule>\
+    </rules>\
+  </ruleset>\
+</patterndb>"
+
+#define pdb_msg_count_skeleton "<patterndb version='4' pub_date='2010-02-22'>\
```

Comment:
If I understand correctly, these were originally gchar * types. Why is these changed to defines?

Reply:
I made the XML strings as a macro because I don't like to create variables in header files. If we want to use the const gchar*  then we should create C file instead of H.
I'd like to keep it as is.